main.tf

```diff
@@ -206,50 +201,45 @@ module "node_pools" {
   tags                         = var.tags
 }
 
-# Module Registry - https://registry.terraform.io/modules/Azure/postgresql/azurerm/2.1.0
-module "postgresql" {
-  source  = "Azure/postgresql/azurerm"
-  version = "2.1.0"
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server
+module "flex_postgresql" {
+  source = "./modules/azurerm_postgresql_flex"
 
-  for_each                     = local.postgres_servers != null ? length(local.postgres_servers) != 0 ? local.postgres_servers : {} : {}
+  for_each = local.postgres_servers != null ? length(local.postgres_servers) != 0 ? local.postgres_servers : {} : {}
 
   resource_group_name          = local.aks_rg.name
   location                     = var.location
-  server_name                  = lower("${var.prefix}-${each.key}-pgsql")
+  server_name                  = lower("${var.prefix}-${each.key}") # suffix '-flexpsql' added in the module
   sku_name                     = each.value.sku_name
   storage_mb                   = each.value.storage_mb
   backup_retention_days        = each.value.backup_retention_days
   geo_redundant_backup_enabled = each.value.geo_redundant_backup_enabled
   administrator_login          = each.value.administrator_login
   administrator_password       = each.value.administrator_password
   server_version               = each.value.server_version
-  ssl_enforcement_enabled      = each.value.ssl_enforcement_enabled
   firewall_rule_prefix         = "${var.prefix}-${each.key}-postgres-firewall-"
   firewall_rules               = local.postgres_firewall_rules
-  vnet_rule_name_prefix        = "${var.prefix}-${each.key}-postgresql-vnet-rule-"
```

Comment:
The vnet elements need to be added back into the code base.

Reply:
`vnet_rule_name_prefix` is not a supported argument for flexible server. See the list of supported arguments [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server#arguments-reference).